apps/aehttp/src/sc_ws_handler.erl

```diff
@@ -68,42 +73,66 @@ websocket_init(Params) ->
     end.
 
 -spec websocket_handle(term(), handler()) -> {ok, handler()}.
-websocket_handle({text, MsgBin}, State) ->
-    try jsx:decode(MsgBin, [return_maps]) of
-        Msg ->
-            case process_incoming(Msg, State) of
-                no_reply -> {ok, State};
-                {reply, Resp} -> {reply, {text, jsx:encode(Resp)}, State};
-                {error, _} -> {ok, State}
+websocket_handle({text, MsgBin}, #handler{protocol = P} = H) ->
+    try unpack_request(P, jsx:decode(MsgBin, [return_maps]), H) of
+        {Msg, H1} ->
+            HNext = H1#handler{orig_request = undefined},
+            try process_incoming(Msg, H1) of
+                no_reply      -> {ok, HNext};
+                {reply, Resp} -> {reply, {text, jsx:encode(Resp)}, HNext}
+            catch
+                error:E1 ->
+                    lager:debug("CAUGHT E1=~p / ~p", [E1, erlang:get_stacktrace()]),
```

Comment:
I see [this stacktrace retrieval not evaluated if debug logs not set](https://github.com/aeternity/lager/blob/3.6.0.ae1/src/lager_transform.erl#L220) so ok.